DEPRECATIONS.md

```diff
@@ -0,0 +1,67 @@
+# Deprecations & Breaking Changes in KEDA
+
+This document describes how KEDA handles deprecations and breaking changes and should provide clarity to KEDA end-users on what to expect when upgrading to a new version of KEDA.
+
+If you are interested in the current open deprecations, we recommend checking our [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations) or the open breaking changes](https://github.com/kedacore/keda/issues?q=is%3Aopen+label%3Abreaking-change+sort%3Aupdated-desc).
+
+## Our versioning strategy
+
+KEDA follows the [Semantic Versioning](https://semver.org/) strategy and can only introduce breaking changes in new major versions.
+
+However, depending on the nature of the change, we may introduce deprecations in existing minor versions. These changes always must be backwards-compatible.
+
+Depending on the area of the change, we might not need to wait for the next major version of KEDA to introduce a breaking change.
```

Comment:
How should I read this?

> These changes always must be backwards-compatible.

Does that mean, that we will support a previous version of the changed stuff forever?

Reply:
If we make a change in a minor version (not CRD based) we must be backwards compatible until next major version.

So if we decide to change something to our Prometheus integration, for example, we can but it should not impact existing functionality by default.

Totally fictious scenario to give an indication - If we introduce `endpointUri` instead of `Uri`, we must support uri until 3.x.

Reply:
I agree that there are some things that we can't delete, but IMO not breaking the compatibility doesn't mean that we will support and add new features to a deprecated thing.
Let's say that we unify prometheus, obviously we should support the old port till a major version, but I wouldn't ensure that the old endpoint will expose new metrics in the future, once a feature is deprecated, I'd keep it as it is till next major, if any user wants the new metrics, they should move to the new endpoint.
WDYT?

Reply:
I think that's something to be determined on a feature-per-feature basis. If the effort is 0 to add it to both, then why not? It's up to them to still choose if they will migrate or not.

However, the sooner they will the better because we will cut it off at some point anyway.

So long story short - How aggressive do we want to be? I'd say case-by-case by not adding to deprecated things should be the norm for sure.

Let me put that in there that that is how it is, and if we still add something to a deprecated feature then no biggy; but we don't doc it

Reply:
Updated after our discussion